File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.java
class Solution {
    public int numSub(String s) {
        long count = 0, c = 0;
        int n = s.length();
        for(int i = 0 ; i<n ; i++){
            if(s.charAt(i) == '1'){
                c++;
            }
            else if(s.charAt(i) == '0'){
                count += (c * (c + 1)) / 2;
                c = 0;
            }
        }
        count += (c * (c + 1)) / 2;
        return (int)(count % 1000000007);
    }
}

File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.cpp
class Solution {
public:
    int numSub(string s) {
        const long long MOD = 1000000007;
        long long res = 0;
        long long cnt = 0; 
        for (char c : s) {
            if (c == '1') {
                cnt++; 
            } else {
                res = (res + (cnt * (cnt + 1) / 2) % MOD) % MOD;
                cnt = 0;
            }
        }
        if (cnt > 0) {
            res = (res + (cnt * (cnt + 1) / 2) % MOD) % MOD;
        }
        
        return (int)res;
    }
};
